RecoJets/JetProducers/plugins/HFJetShowerShape.cc

```diff
@@ -0,0 +1,240 @@
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "TMath.h"
+#include <iostream>
+
+//
+// class declaration
+//
+
+class HFJetShowerShape : public edm::stream::EDProducer<> {
+public:
+  explicit HFJetShowerShape(const edm::ParameterSet&);
+  ~HFJetShowerShape() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  template <typename T>
+  void putInEvent(const std::string&, const edm::Handle<edm::View<reco::Jet>>&, std::vector<T>*, edm::Event&) const;
+
+  edm::EDGetTokenT<edm::View<reco::Jet>> jets_token_;
+  edm::EDGetTokenT<std::vector<reco::Vertex>> vertices_token_;
+
+  //Jet pt/eta thresholds
+  double jetPtThreshold_, jetEtaThreshold_;
+  //HF geometry
+  double hfTowerEtaWidth_, hfTowerPhiWidth_;
+  //Variables for PU subtraction
+  double vertexRecoEffcy_, offsetPerPU_, jetReferenceRadius_;
+  //Pt thresholds for showershape variable calculation
+  double stripPtThreshold_, widthPtThreshold_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+HFJetShowerShape::HFJetShowerShape(const edm::ParameterSet& iConfig) {
+  jets_token_ = consumes<edm::View<reco::Jet>>(iConfig.getParameter<edm::InputTag>("theJets"));
+  vertices_token_ = consumes<std::vector<reco::Vertex>>(iConfig.getParameter<edm::InputTag>("theVertices"));
+  jetPtThreshold_ = iConfig.getParameter<double>("jetPtThreshold");
+  jetEtaThreshold_ = iConfig.getParameter<double>("jetEtaThreshold");
+  hfTowerEtaWidth_ = iConfig.getParameter<double>("hfTowerEtaWidth");
+  hfTowerPhiWidth_ = iConfig.getParameter<double>("hfTowerPhiWidth");
+  vertexRecoEffcy_ = iConfig.getParameter<double>("vertexRecoEffcy");
+  offsetPerPU_ = iConfig.getParameter<double>("offsetPerPU");
+  jetReferenceRadius_ = iConfig.getParameter<double>("jetReferenceRadius");
+  stripPtThreshold_ = iConfig.getParameter<double>("stripPtThreshold");
+  widthPtThreshold_ = iConfig.getParameter<double>("widthPtThreshold");
+
+  produces<edm::ValueMap<float>>("sigmaEtaEta");
+  produces<edm::ValueMap<float>>("sigmaPhiPhi");
+  produces<edm::ValueMap<int>>("centralEtaStripSize");
+  produces<edm::ValueMap<int>>("adjacentEtaStripsSize");
+}
+
+HFJetShowerShape::~HFJetShowerShape() {
+  // do anything here that needs to be done at destruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void HFJetShowerShape::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+
+  //Jets
+  edm::Handle<edm::View<reco::Jet>> theJets;
+  iEvent.getByToken(jets_token_, theJets);
+
+  //Vertices
+  edm::Handle<std::vector<reco::Vertex>> theVertices;
+  iEvent.getByToken(vertices_token_, theVertices);
+  int nPV = theVertices->size();
```

Comment:
```suggestion
  int nPV = iEvent.get(vertices_token_).size();
```
it looks like only size is needed